torch/distributed/rpc/api.py

```diff
@@ -259,22 +259,22 @@ def remote(to, func, args=None, kwargs=None):
         >>> rpc.shutdown()
     """
     qualified_name = torch.jit._find_builtin(func)
-
-    args = args if args else ()
-    kwargs = kwargs if kwargs else {}
-
     info = _to_worker_info(to)
-    current_worker = get_worker_info()
+
     # If profiling is enabled, kick off the timer and retrieve back a
     # RecordFunction instance.
     rf = None
     if torch.autograd._profiler_enabled():
         rf = _start_record_function(
             RPCExecMode.REMOTE,
-            str(qualified_name if qualified_name is not None else func.__name__),
-            current_worker.name,
+            str(qualified_name) if qualified_name is not None else func.__qualname__,
+            get_worker_info().name,
             info.name,
         )
+
+    args = args if args else ()
+    kwargs = kwargs if kwargs else {}
```

Comment:
Why do we need these additional lines?

Reply:
It looks like this was moved in an earlier version of this PR. It shouldn't have to be moved.